File: src/easy/ReverseLinkedList.swift
func reverseLinkedList<T: Equatable>(_ head: ListNode<T>?) -> ListNode<T>? {
  var previous: ListNode<T>? = nil
  var next: ListNode<T>? = nil
  var current: ListNode<T>? = head
  while current != nil {
    next = current?.next
    current?.next = previous
    previous = current
    current = next
  }
  return previous
}

File: src/main/php/Easy/ReverseLinkedList.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class ReverseLinkedList
{
    public function solve(?ListNode $head): ?ListNode
    {
        $previous = null;
        $current = $head;
        while ($current !== null) {
            $next = $current->next;
            $current->next = $previous;
            $previous = $current;
            $current = $next;
        }

        return $previous;
    }
}
